src/annotator/communicator.js

```diff
@@ -0,0 +1,375 @@
+import { ListenerCollection } from './util/listener-collection';
+
+/**
+ * @typedef {import('../types/config').SidebarConfig} SidebarConfig
+ * @typedef {import('../types/annotator').Destroyable} Destroyable
+ *
+ * @typedef Message
+ * @prop {'guestToSidebarChannel'|'hostToSidebarChannel'|'notebookToSidebarChannel'} channel
+ * @prop {'guest'|'notebook'|'sidebar'} port
+ * @prop {'offer'|'request'}  type
+ *
+ * @typedef {Message & {source: 'hypothesis'}} MessageSourced
+ */
+
+// Because there are many `postMessages` in the wild, this PREFIX helps to
+// identify the provenance and avoids the chances of cross-communication
```

Comment:
I wanted the source property to be use as a guard to differentiate `hypothesis` postMessages from other postMessages. We don't want to react to other postMessages that could have the same format `{channel, port, type}`, although it is unlikely. 

It is not a security feature but short of an anti-collision mechanism.

I will modify the comment.

Reply:
> It is not a security feature but short of an anti-collision mechanism.

I see. So this is intended to give us increased confidence that a message is from us rather than someone else. It doesn't help third-party code misinterpreting our messages, assuming they didn't have the foresight to add a `source` key and check it. If we ended up breaking other sites with these messages in a significant way we'd have to change things on our end.